### DISCUSSION TITLE
Unrecoverable error

### DISCUSSION THREAD

**Discussion by BugPhobic** [firefly-iii/firefly-iii#6570] [Firefly III] [answered]
I loved what FireFly offered, so I learned the basics of self hosting and set up a Yunohost server using VirtualBox virtual machine. Things seemed to go alright until I started to get error: file_put_contents(/var/www/firefly-iii/storage/framework/cache/data/22/b3/22b39ddfe6fdc55f1d3fdcc9f96d84fffa4b454e): Failed to open stream: No such file or directory

I first got this error when I added some notes to a group or expense/revenue account. I found that going back a page and creating a new expense/revenue account fixed the issue. 

Slowly the frequency of the error started to increase.

Finally, after I made an update to a budget and clicked the update button, I got this error. No amount of back-tabbing worked, and I got the error all the way to the Firefly dashboard. 

I logged out, shut down the virtual machine, and restarted it. Cleared the browser cookies and history too. Logged into YunoHost user portal no problem. Firefly shows its login page no problem. But once I log in, instead of the dashboard, I get this error.

I have no clue about how to get access to my database from the Yunohost server. The virtual machine's data is stored in .vdi file, if that matters.

Is this an issue with the virtual machine storage file being corrupt? If so, how do I retrieve the data that I did save? I only spent 5 hours using Firefly before it became unusable, but I would prefer not having to setup 50 categories again.

Your help and advice will be greatly appreciated!

I don't know how to get access to the logs in storage either. Can someone guide me? Perhaps the logs there might be more useful.

Here's the debug info:

Debug information generated at 2022-10-25 21:06:55 Canada/Eastern for Firefly III version **5.7.12**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.12 |
| Firefly III API | 1.5.6 |
| Build | (unknown), (unknown) |
| DB version | 18 (exp. 18) |
| PHP | `8.0.24` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Canada/Eastern |
| Browser TZ | America/New_York |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql |
| Current driver | mysql |
| Trusted proxies (.env) | `` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-10-01 00:00:00 |
| Session end | 2022-10-31 23:59:59 |
| Session first | 2022-09-30 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_CA |
| Attempt at "en_CA.utf8" | 'en_CA.utf8' |
| Attempt at "en_CA.UTF-8" | 'en_CA.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36 |

**Comment by BugPhobic** [comment, marked as answer]
> I loved what FireFly offered, so I learned the basics of self hosting and set up a Yunohost server using VirtualBox virtual machine. Things seemed to go alright until I started to get error: file_put_contents(/var/www/firefly-iii/storage/framework/cache/data/22/b3/22b39ddfe6fdc55f1d3fdcc9f96d84fffa4b454e): Failed to open stream: No such file or directory
> 
> I first got this error when I added some notes to a group or expense/revenue account. I found that going back a page and creating a new expense/revenue account fixed the issue. 
> 
> Slowly the frequency of the error started to increase.
> 
> Finally, after I made an update to a budget and clicked the update button, I got this error. No amount of back-tabbing worked, and I got the error all the way to the Firefly dashboard. 
> 
> I logged out, shut down the virtual machine, and restarted it. Cleared the browser cookies and history too. Logged into YunoHost user portal no problem. Firefly shows its login page no problem. But once I log in, instead of the dashboard, I get this error.
> 
> I have no clue about how to get access to my database from the Yunohost server. The virtual machine's data is stored in .vdi file, if that matters.
> 
> Is this an issue with the virtual machine storage file being corrupt? If so, how do I retrieve the data that I did save? I only spent 5 hours using Firefly before it became unusable, but I would prefer not having to setup 50 categories again.
> 
> Your help and advice will be greatly appreciated!
> 
> I don't know how to get access to the logs in storage either. Can someone guide me? Perhaps the logs there might be more useful.
> 
> Here's the debug info:
> 
> Debug information generated at 2022-10-25 21:06:55 Canada/Eastern for Firefly III version **5.7.12**.
> 
> | Scope | Version |
> | --- | --- |
> | Firefly III | 5.7.12 |
> | Firefly III API | 1.5.6 |
> | Build | (unknown), (unknown) |
> | DB version | 18 (exp. 18) |
> | PHP | `8.0.24` |
> | Host | `Linux` |
> 
> | System info | Value |
> | --- | --- |
> | System TZ | Canada/Eastern |
> | Browser TZ | America/New_York |
> | App environment | local |
> | App debug mode | false |
> | App cache driver | file |
> | App logging | notice, stack |
> | Display errors | Off |
> | Error reporting | ALL errors |
> | Interface | fpm-fcgi |
> | Default language | en_US |
> | Default locale | equal |
> | BCscale | 24 |
> | DB drivers | mysql |
> | Current driver | mysql |
> | Trusted proxies (.env) | `` |
> 
> | User info | Value |
> | --- | --- |
> | Login provider | eloquent  |
> | User guard | web  |
> | Headers | N/A, N/A  |
> | Stateful domains |  |
> | Session start | 2022-10-01 00:00:00 |
> | Session end | 2022-10-31 23:59:59 |
> | Session first | 2022-09-30 00:00:00 |
> | User ID | 1 |
> | User language | en_US |
> | User locale | en_CA |
> | Attempt at "en_CA.utf8" | 'en_CA.utf8' |
> | Attempt at "en_CA.UTF-8" | 'en_CA.UTF-8' |
> | User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36 |
> 

EDIT: By going to domainname.com/firefly/flush, I was able to remove the error and login to my dashboard.

But is there a reason why I keep getting this error?

**Reply by JC5** [reply]
> But is there a reason why I keep getting this error?

https://docs.firefly-iii.org/firefly-iii/faq/self_hosted/#unable-to-write-to-cache-directory

**Reply by BugPhobic** [reply]
Thanks! I have yet to see the error since I tried the flush URL though. Maybe that fixed it for me. Been working on it for another 3 hours adding lots of transactions and rules and not a single error so far, even though I only tried the flush thing.

Just two quick questions while you're here:

Is it possible to mark a transfer as a bill? I would love to keep track of my upcoming credit card bills and see if I paid them as some of my credit cards are manually paid. Having an all-in-one bill tracker would be amazing. The one in Firefly is already very powerful!

I know you stated you will not be making further upgrades to the rule engine to prevent it from getting overly complicated, but will you be adding OR and AND functions to the search (not rules)? That would be amazing!
